hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/HoodieFlinkWriteClient.java

```diff
@@ -369,7 +369,8 @@ public void completeCompaction(
       // commit to data table after committing to metadata table.
       // Do not do any conflict resolution here as we do with regular writes. We take the lock here to ensure all writes to metadata table happens within a
       // single lock (single writer). Because more than one write to metadata table will result in conflicts since all of them updates the same partition.
-      table.getMetadataWriter().ifPresent(w -> w.update(metadata, compactionInstant.getTimestamp(), table.isTableServiceAction(compactionInstant.getAction())));
+      table.getMetadataWriter(compactionInstant.getTimestamp()).ifPresent(
```

Comment:
Have filed a ticket to fix Flink around metadata table instantiation 
https://issues.apache.org/jira/browse/HUDI-2866
I don't see concurrency support in flink and so may not be very much needed as we need in spark, but anyways.

Reply:
cc @danny0405 FYI